tools/ros2_dependencies.repos

```diff
@@ -0,0 +1,5 @@
+repositories:
```

Comment:
@naiveHobo I know I am coming late to the party, but why did we add this dependency? Is it because of the costmaps? @na

Reply:
Okay, I guess because of 

```
I think the best answer is to add a tools/ros2_dependencies.repos file that contains the vcs address for map server so its built in the same workspace (isolate / remove all other packages from nav2 that aren't needed for map server)
```

Reply:
Yes, we need nav2_costmap_2d as a dependency but its not yet released since its super high in the dependency tree. Grid maps will at some point be also a dependency of navigation2 which creates a circular dependency so both nav2 and grid maps need to build certain elements of each other in CI to deal with a chicken and egg problem. Also lets both build master branch in case there are API breaking changes we can fix them more quickly

Reply:
Okay, sounds good. Thanks.